src/main_prog.py

```diff
@@ -0,0 +1,47 @@
+import argparse
+
+import material_properties
+
+import generate_mesh
+
+# Allow for user input for number of materials and what kind of materials
+
+parser = argparse.ArgumentParser()
+    
+parser.add_argument("-n", "--number_materials", type=int, default=3, required=False,
```

Comment:
Is this necessary given the other inputs?  What if they don't match?   Can you just determine this based by counting the number of material specified in the other arguments?

Reply:
Good point, I don't think this is needed.